core/chains/sui.go

```diff
@@ -0,0 +1,179 @@
+			NextEpochP2PAddress          interface{} `json:"nextEpochP2pAddress"`
+			NextEpochPrimaryAddress      interface{} `json:"nextEpochPrimaryAddress"`
+			NextEpochWorkerAddress       interface{} `json:"nextEpochWorkerAddress"`
+			VotingPower                  string      `json:"votingPower"`
+			OperationCapID               string      `json:"operationCapId"`
+			GasPrice                     string      `json:"gasPrice"`
+			CommissionRate               string      `json:"commissionRate"`
+			NextEpochStake               string      `json:"nextEpochStake"`
+			NextEpochGasPrice            string      `json:"nextEpochGasPrice"`
+			NextEpochCommissionRate      string      `json:"nextEpochCommissionRate"`
+			StakingPoolID                string      `json:"stakingPoolId"`
+			StakingPoolActivationEpoch   string      `json:"stakingPoolActivationEpoch"`
+			StakingPoolDeactivationEpoch interface{} `json:"stakingPoolDeactivationEpoch"`
+			StakingPoolSuiBalance        string      `json:"stakingPoolSuiBalance"`
+			RewardsPool                  string      `json:"rewardsPool"`
+			PoolTokenBalance             string      `json:"poolTokenBalance"`
+			PendingStake                 string      `json:"pendingStake"`
+			PendingTotalSuiWithdraw      string      `json:"pendingTotalSuiWithdraw"`
+			PendingPoolTokenWithdraw     string      `json:"pendingPoolTokenWithdraw"`
+			ExchangeRatesID              string      `json:"exchangeRatesId"`
+			ExchangeRatesSize            string      `json:"exchangeRatesSize"`
+		} `json:"activeValidators"`
+		PendingActiveValidatorsID   string          `json:"pendingActiveValidatorsId"`
+		PendingActiveValidatorsSize string          `json:"pendingActiveValidatorsSize"`
+		PendingRemovals             []interface{}   `json:"pendingRemovals"`
+		StakingPoolMappingsID       string          `json:"stakingPoolMappingsId"`
+		StakingPoolMappingsSize     string          `json:"stakingPoolMappingsSize"`
+		InactivePoolsID             string          `json:"inactivePoolsId"`
+		InactivePoolsSize           string          `json:"inactivePoolsSize"`
+		ValidatorCandidatesID       string          `json:"validatorCandidatesId"`
+		ValidatorCandidatesSize     string          `json:"validatorCandidatesSize"`
+		AtRiskValidators            []interface{}   `json:"atRiskValidators"`
+		ValidatorReportRecords      [][]interface{} `json:"validatorReportRecords"`
+	} `json:"result"`
+	ID int `json:"id"`
+}
+
+type rawBody struct {
+		JSONRPC string        `json:"jsonrpc"`
+		ID      int           `json:"id"`
+		Method  string        `json:"method"`
+		Params  []interface{} `json:"params"`
+	}
+
+func Sui() (int, error) {
+	var votingPowers []big.Int
+
+	url := "https://fullnode.mainnet.sui.io"
+
+	request := rawBody {
+		JSONRPC: "2.0",
+		ID:      1,
+		Method:  "suix_getLatestSuiSystemState",
+		Params:  []interface{}{},
+	}
+
+	requestBody, err := json.Marshal(request)
```

Comment:
can we align this code on how we make API requests for other chains in this repository? For example, check cosmos